source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -521,7 +524,68 @@ void RtCoreIo::read(const DocumentPtr& doc, RtCoreIo::ReadFilter filter)
     }
 }
 
-void RtCoreIo::write(DocumentPtr& doc, RtCoreIo::WriteFilter filter)
+void RtFileIo::read(const FilePath& documentPath, const FileSearchPath& searchPaths, RtFileIo::ReadFilter filter)
+{
+    try
+    {
+        DocumentPtr document = createDocument();
+        XmlReadOptions readOptions;
+        readOptions.skipConflictingElements = true;
+        readFromXmlFile(document, documentPath, searchPaths, &readOptions);
+
+        CopyOptions copyOptions;
+        copyOptions.skipConflictingElements = true;
+
+        read(document, filter);
+    }
+    catch(Exception&)
+    {
+        return;
+    }
+}
+
+void RtFileIo::loadLibraries(const StringVec& libraryPaths, const FileSearchPath& searchPaths, const RtToken& referenceName)
+{
+    // Always add in local search paths
+    FileSearchPath totalSearchPaths = searchPaths;
+    FilePath currentPath(FilePath::getCurrentPath());
+    FilePath parentCurrentPath = currentPath.getParentPath();
+    const FilePath librariesPath("libraries");
```

Comment:
I'm not sure we should add in implied knowledge of where library files are located to the general API. I think it's better to handle this on client side, adding to the serachPaths before calling this method. Otherwise we risk introducing unexpected behavior for clients that use different directories for their libraries.

Reply:
Agreed.